proxy/proxy.go

```diff
@@ -30,6 +35,17 @@ func MiddlewareSetUpstream(upstream *url.URL) Middleware {
 	}
 }
 
+func MiddlewareSetPrefixHeader(headerKey string) Middleware {
+	return func(r *http.Request) {
+		prefix, ok := GetTenantPrefix(r.Context())
```

Comment:
This really has nothing to do with the tenant, right? It's just a prefix used by the proxy. In fact, the middleware is called `SetPrefixHeader`. Maybe we can make this consistent and remove `Tenant`